Doc updates for JF 10.8 HWA improvements (#595)
jellyfin/jellyfin-docs: general/administration/hardware-acceleration.md

Linux   | QSV, NVENC, AMF, VA-API
Windows | QSV, NVENC, AMF
MacOS   | VideoToolbox
RPi     | V4L2, OMX (deprecated)

[Graphics Cards comparison using HWA](https://www.elpamsoft.com/?p=Plex-Hardware-Transcoding)

Based on hardware vendor:

Vendor  | Supported HW Acceleration
------- | -------------
NVIDIA  | NVENC
AMD     | AMF, VA-API
Intel   | QSV, VA-API
Apple   | VideoToolbox

> RPi | V4L2, OMX (deprecated)

Reviewer: ```suggestion
RPi    | V4L2, OMX (deprecated)
```

lshw -C display
```

## Tips on H.264 / AVC 10-bit videos

The hardware decoding of H.264 10-bit aka High10 profile video is not supported by any Intel, AMD or NVIDIA GPU.

Please consider upgrading these videos to HEVC 10-bit aka Main10 profile if you want to offload your CPU usage during transcoding.

## Tips for Intel Gen9 and Gen11+ when using VAAPI or QSV on Linux

The Intel [Guc/Huc firmware](https://wiki.archlinux.org/title/intel_graphics#Enable_GuC_/_HuC_firmware_loading) must be enabled for optional Low-Power encoding (pre-Gen11 only supports LP H.264).

> 

Reviewer: We should add some documentation or links on how to enable it on Debian/Ubuntu based distros. The Arch wiki isn't useful in this cases.

lshw -C display
```

## Tips on H.264 / AVC 10-bit videos

The hardware decoding of H.264 10-bit aka High10 profile video is not supported by any Intel, AMD or NVIDIA GPU.

Please consider upgrading these videos to HEVC 10-bit aka Main10 profile if you want to offload your CPU usage during transcoding.

## Tips for Intel Gen9 and Gen11+ when using VAAPI or QSV on Linux

The Intel [Guc/Huc firmware](https://01.org/linuxgraphics/downloads/firmware) must be enabled for optional Low-Power encoding (pre-Gen11 only supports LP H.264).

Instructions for ArchLinux: [Arch Wiki](https://wiki.archlinux.org/title/intel_graphics#Enable_GuC_/_HuC_firmware_loading) and for Debian/Ubuntu: [Brainiarc7's gist](https://gist.github.com/Brainiarc7/aa43570f512906e882ad6cdd835efe57)

For Jasper Lake chips (such as N5095 and N6005), Low-Power encoding must be enabled. But the linux-firmware support is not ready in Ubuntu 20.04.3 LTS. The 21.10 or newer distro is required.

## Supported Acceleration Methods

> Since Jellyfin 10.8 release, the full hardware filtering (scale, de-interlacle, tone-mapping and subtitle burn-in) on Intel, AMD and NVIDIA have been verified with `jellyfin-ffmpeg` 4.4.1.

Reviewer 2: I think `since` can confuse people in the future in thinking that lets say, 10.9 has been tested with `jellyfin-ffmpeg` 4.4.1.
```suggestion
In Jellyfin 10.8, full hardware-accelerated filtering (scaling, deinterlacing, tone mapping and subtitle burn-in) on Intel, AMD and NVIDIA hardware have been verified with `jellyfin-ffmpeg` 4.4.1.
```

1. On Windows: Install the latest NVIDIA or AMD or Intel drivers.

2. On Linux or docker:
   - For AMD cards, install `amdgpu-pro` with opencl arguments. See **Configuring AMD AMF encoding on Ubuntu 18.04 or 20.04 LTS** for more details.

    ```sh
    sudo ./amdgpu-pro-install -y --opencl=pal,legacy
    sudo usermod -aG video $LOGNAME
    sudo usermod -aG render $LOGNAME
    ```

   - For Intel iGPUs, you have two types of tonemapping methods: OpenCL and VPP. Choose the latter one for faster transcoding speed, but fine tuning options are not supported.

    Method OpenCL: Follow the instructions from [intel-compute-runtime](https://github.com/intel/compute-runtime/releases). If you are using the docker image from jellyfin/jellyfin or linuxserver/jellyfin, this step can be skipped.

    Method VPP: Install `intel-media-va-driver-non-free` 20.1 and `jellyfin-ffmpeg` 4.4.1-1 or newer.

   > [!NOTE]
   > Tone mapping on Intel VA-API and QSV needs an iGPU that supports 10-bit decoding, such as i3-7100 and J4105.
   > Do not use the `intel-opencl-icd` package from the repository since they were not build with RELEASE_WITH_REGKEYS enabled for P010 pixel interop flags.

> 3. Debuging: Check the OpenCL device status. You will see corresponding vendor name if it goes well.

Reviewer 3: ```suggestion
3. Debugging: Check the OpenCL device status. You will see corresponding vendor name if it goes well.
```

Intel hardware based VPP tone-mapping is supported on Intel QSV and VA-API on Linux. VPP is prefered when both two tone-mapping options are checked on Intel.

OS/Platform | NVIDIA NVENC | AMD AMF  | Intel QSV | Intel VA-API | AMD VA-API | Software
----------- | ------------ | -------- | ----------- | --------- | --------- | --------
Linux       | ✔️          | ✔️       | ✔️        | ✔️          | ✔️   | WIP
Windows     | ✔️           | ✔️       | ✔️        | N/A         | N/A       | WIP
Docker      | ✔️           | untested | ✔️        | ✔️          | untested   | WIP

> [!NOTE]
> Tone-mapping on Windows with Intel QSV and AMD AMF requires Windows 10 or newer.

> [!NOTE]
> Make sure the hardware acceleration is well configured before reading this section.

1. On Windows: Install the latest NVIDIA or AMD or Intel drivers.

2. On Linux or docker:
   - For AMD cards, install `amdgpu-pro` with opencl arguments. See **Configuring AMD AMF encoding on Ubuntu 18.04 or 20.04 LTS** for more details.

    ```sh
    sudo ./amdgpu-pro-install -y --opencl=pal,legacy
    sudo usermod -aG video $LOGNAME

>     sudo usermod -aG render $LOGNAME

Reviewer: What should be substituted for `$LOGNAME`?

Author: This was copied from AMD's doc. LOGNAME => the name of the login user

Reviewer: Should we calarify this or just keep it as is since the AMD docs need to be consulted anyway?

Author: Leave it as is. LOGNAME is a linux environment variable.
https://en.wikipedia.org/wiki/Logname

    ```

>    - For Intel iGPUs, you have two types of tonemapping methods: OpenCL and VPP. Choose the latter one for faster transcoding speed, but fine tuning options are not supported.

Reviewer: ```suggestion
   - For Intel iGPUs, you have two types of tone-mapping methods: OpenCL and VPP. Choose the latter one for faster transcoding speed, but fine tuning options are not supported.
```
